osu.Game/Database/StandardisedScoreMigrationTools.cs

```diff
@@ -264,9 +266,65 @@ public static long ConvertFromLegacyTotalScore(ScoreInfo score, LegacyBeatmapCon
             switch (score.Ruleset.OnlineID)
             {
                 case 0:
+                    if (score.MaxCombo == 0 || score.Accuracy == 0)
+                    {
+                        return (long)Math.Round((
+                            0
+                            + 500000 * Math.Pow(score.Accuracy, 5)
+                            + bonusProportion) * modMultiplier);
+                    }
+
+                    // Assumption :
+                    // - sliders and slider-ticks are uniformly spread arround the beatmap, and thus can be ignored without losing much precision.
+                    //      We thus consider a map of hit-circles only, which gives objectCount == maximumCombo !
+                    // - the Ok/Meh hit results are uniformly spread in the score, and thus can be ignored without losing much precision.
+                    //      We simplify and consider each hit result to be equal to `300*score.Accuracy`, which allows us to isolate the accuracy multiplier.
+                    // What is strippedV1/strippedNew :
+                    // This is the ComboScore (of v1 / new score) were we remove all (map-)constant multipliers and accuracy multipliers (including hit results),
+                    // based on the previous assumptions.
+                    // We use integrals to approximate the sum of each object's combo contribution (thus the original combo exponent is increased by 1)
+                    // For the maximum score, we thus integrate `f(combo) = 300*combo^exponent`, and ignoring the constant multipliers we get:
+                    double maxStrippedV1 = Math.Pow(maximumLegacyCombo, 2);
+                    double maxStrippedNew = Math.Pow(maximumLegacyCombo, 1 + ScoreProcessor.COMBO_EXPONENT);
+
+                    double strippedV1 = maxStrippedV1 * comboProportion / score.Accuracy;
```

Comment:
>  Is that part supposed to somehow undo or counteract the accuracy part of the score or whatever ?
Yes, because the ComboScore of v1 includes accuracy (it's the hit-results values that are multiplied by the current combo at each step).

> I guess https://github.com/ppy/osu/pull/24166#issuecomment-1821785569 seems related?
No, it's not.